File: app/src/main/java/com/mstc/mstcapp/ui/resources/detailsTab/DetailsViewModel.java
package com.mstc.mstcapp.ui.resources.detailsTab;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.mstc.mstcapp.Repository;
import com.mstc.mstcapp.model.resources.DetailModel;

public class DetailsViewModel extends AndroidViewModel {
    Repository repository;
    LiveData<DetailModel> liveData;

    public DetailsViewModel(@NonNull Application application) {
        super(application);
        repository = new Repository(application);
    }

    public LiveData<DetailModel> getDetails(String domain) {
        liveData = repository.getDetails(domain);
        return liveData;
    }

    public void insertDetails(String domain, DetailModel details) {
        repository.insertDetails(domain, details);
    }
}

File: app/src/main/java/com/mstc/mstcapp/ui/resources/details/DetailsViewModel.kt
package com.mstc.mstcapp.ui.resources.details

import android.app.Application
import android.content.Context
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import com.mstc.mstcapp.data.ResourceRepository
import com.mstc.mstcapp.data.STCDatabase
import com.mstc.mstcapp.model.Result
import com.mstc.mstcapp.model.resource.Detail
import kotlinx.coroutines.launch

class DetailsViewModel(application: Application) : AndroidViewModel(application) {

    val context: Context by lazy { application.applicationContext }
    private val repository =
        ResourceRepository(context, STCDatabase.getInstance(context))

    fun getDetails(domain: String): LiveData<Result<Detail>> {
        return repository.getDomainDetails(domain)
    }

    fun refreshDetails(domain: String) {
        viewModelScope.launch {
            repository.refreshDetails(domain)
        }
    }

}